### PR TITLE
docs: migration for hooks

### PR DESCRIPTION
**Pull request by prince-chrismc**
I stole this from #2877

### CODE REVIEW

File: migrating_to_2.0/general.rst [hunk 1]
```diff
@@ -54,6 +54,15 @@ Extensions
 
 Work in progress
 
+Hooks
+^^^^^
+
+- Hooks folder has been updated to ``~/.conan2.extensions/hook``;
+- Any hook file must be named with ``hook_`` as prefix and ``.py`` as suffix;
+- Only ``ConanFile`` is passed as parameter;
+- Pre and Post Download are no longer supported in Conan 2.x
```
Comment (SSE4) [inline migrating_to_2.0/general.rst:63]: what's the replacement?

File: migrating_to_2.0/general.rst [hunk 2]
```diff
@@ -54,6 +54,15 @@ Extensions
 
 Work in progress
 
+Hooks
+^^^^^
+
+- Hooks folder has been updated to ``~/.conan2.extensions/hook``;
```
Comment (SSE4) [inline migrating_to_2.0/general.rst:60]: is it really `.conan2.extension/hook`, not `.conan2/extensions/hooks`?

Comment (prince-chrismc) [reply migrating_to_2.0/general.rst:60]:
```suggestion
- Hooks folder has been updated to ``~/.conan2/extensions/hook``;
```

Comment (prince-chrismc) [reply migrating_to_2.0/general.rst]: Good catch!

File: migrating_to_2.0/general.rst (hunk 1 above)
Comment (prince-chrismc) [reply migrating_to_2.0/general.rst:63]: I do not know the significance, I just stole this from a PR to help :)

Comment (memsharded) [reply migrating_to_2.0/general.rst:63]: There are no replacement. Hooks are recipe pre/post methods.
Package signing checks happens post-download too. Other cases need to be discussed, but they will be plugins, not hooks (not necessarily having ``conanfile`` as an argument)

**Review by memsharded — APPROVED** [review]

File: migrating_to_2.0/general.rst (hunk 1 above)
Comment (prince-chrismc) [reply migrating_to_2.0/general.rst:63]: Thanks for the details 👍

**Pull request merged**